web-app/src/components/Commentary/Dialog.js

```diff
@@ -0,0 +1,79 @@
+import React, { useContext } from 'react';
```

Comment:
This file is not yet complete as this is for the metadata, you can finish it in the next PR as discussed. Will see the final finished logic